tales_query.go

```diff
@@ -58,38 +72,142 @@ func (l *Service) queryDay(ctx context.Context, actors []uint32, day time.Time,
+	// Filter relevant chunks first
+	var tasks []async.Task
+	var taskIndices []int // Track the original index for each task to maintain order
+	for i, chunk := range meta.Chunks {
 		// Skip chunks that don't overlap with query time range
 		if !chunk.Between(fromSec, toSec) {
 			continue
 		}
+		
+		// Wrap the original chunk processing logic in a lambda
 		chunkKey := keyOfChunk(seq.FormatDate(day), chunk.Offset())
-		var index *sroar.Bitmap
-		for _, a := range actors {
-			idx, ok := chunk.Actors[a]
-			if !ok || uint32(idx[0]) < t0 || uint32(idx[0]) > t1 {
-				index = nil
-				break
+		index := i // Capture index for closure
+		
+		task := async.NewTask(func(ctx context.Context) (any, error) {
+			var idx *sroar.Bitmap
+			for _, a := range actors {
+				idxEntry, ok := chunk.Actors[a]
+				if !ok || uint32(idxEntry[0]) < t0 || uint32(idxEntry[0]) > t1 {
+					idx = nil
+					break
+				}
+
+				bitmap, err := l.loadBitmap(ctx, chunkKey, idxEntry)
+				if err != nil {
+					return nil, err
+				}
+
+				switch {
+				case idx == nil:
+					idx = bitmap
+				case !idx.IsEmpty():
+					idx.And(bitmap)
+				}
+			}
+
+			// Collect log entries instead of calling yield
+			if idx != nil && !idx.IsEmpty() {
+				entries, err := l.collectChunkEntries(ctx, chunkKey, chunk, *idx, day, from, to)
+				return chunkTaskResult{
+					index:   index,
+					entries: entries,
+					err:     err,
+				}, nil
+			}
+			return chunkTaskResult{
+				index:   index,
+				entries: nil,
+				err:     nil,
+			}, nil
+		})
+		
+		tasks = append(tasks, task)
+		taskIndices = append(taskIndices, i)
+	}
+
+	if len(tasks) == 0 {
+		return true
+	}
+
+	// Process chunks in parallel using async.Consume
+	return l.processChunksWithAsync(ctx, tasks, yield)
+}
+
+// processChunksWithAsync processes chunks using kelindar/async for task management
+func (l *Service) processChunksWithAsync(ctx context.Context, tasks []async.Task, yield func(time.Time, string) bool) bool {
+	if len(tasks) == 0 {
+		return true
+	}
+
+	// Use sequential processing if only one task or parallel downloads is 1
+	if len(tasks) == 1 || l.config.ParallelDownloads == 1 {
+		task := tasks[0].Run(ctx)
+		outcome, err := task.Outcome()
+		if err != nil {
+			return true // Skip chunks that fail to process
+		}
+		
+		if result, ok := outcome.(chunkTaskResult); ok && result.err == nil {
+			for _, entry := range result.entries {
+				if !yield(entry.timestamp, entry.text) {
+					return false
+				}
+			}
+		}
+		return true
+	}
+
+	// Create a channel to feed tasks to the consumer
+	taskChan := make(chan async.Task, len(tasks))
+	
+	// Send all tasks to the channel
+	go func() {
+		defer close(taskChan)
+		for _, task := range tasks {
+			select {
+			case taskChan <- task:
+			case <-ctx.Done():
+				return
 			}
+		}
+	}()
+
+	// Use async.Consume to process tasks with specified concurrency
+	consumerTask := async.Consume(ctx, l.config.ParallelDownloads, taskChan)
```

Comment:
This should just be started together with the `Service`, it creates a single working pool